src/lib/index/publish.js

```diff
@@ -27,25 +27,77 @@
 'use strict';
 
 const
-	Amqp = require('./shared/amqp'),
+	_ = require('lodash'),
+	amqp = require('amqplib'),
+	{ validate } = require('./shared/configValidator'),
 
 	EventEmitter = require('events'),
 
 	ERROR_EVENT = 'error_event',
 
 	emitter = new EventEmitter(),
 
-	send = ({ eventName, buffer, config }) => {
-		return Amqp.apply(channel => channel.sendToQueue(eventName, buffer), config)
-			.catch(err => {
-				emitter.emit(ERROR_EVENT, err.message);
-				throw err;
-			});
-	};
+	publishers = new Map();
+
+class Publisher {
+
+	constructor(config) {
+		validate(config);
+		this.config = _.cloneDeep(config);
+	}
+
+	async init() {
+		this.connection = await amqp.connect(this.config.cloudamqpUrl);
+		this.channel = await this.connection.createChannel();
+	}
+
+	async publish(queue, buffer) {
+		return await this.channel.sendToQueue(queue, buffer);
```

Comment:
Await isn't necessary, as there's no processing after the Promise resolves. May as well return the Promise directly.
Likewise, async isn't necessary, but perhaps it makes it clearer the function is returning a Promise.

Reply:
Yes I was using it as an indication that this returns a promise.